stan/math/prim/fun/modified_bessel_first_kind.hpp

```diff
@@ -40,6 +40,12 @@ inline T2 modified_bessel_first_kind(int v, const T2 z) {
   return boost::math::cyl_bessel_i(v, z);
 }
 
+inline double modified_bessel_first_kind(int v, int z) {
+  check_not_nan("modified_bessel_first_kind", "z", z);
+
+  return boost::math::cyl_bessel_i(v, double(z));
+}
+
```

Comment:
Why the need for a specialization here? I think you can just change the first one to

```cpp
template <typename T2, require_arithmetic_t<T2>* = nullptr>
inline auto modified_bessel_first_kind(int v, const T2 z) {
  check_not_nan("modified_bessel_first_kind", "z", z);
  return boost::math::cyl_bessel_i(v, static_cast<double>(z));
}
```

Reply:
@pgree was talking to me about this yesterday.

Apparently ```boost::math::cyl_bessel_i``` explodes if the second argument is an int, but it's fine to cast it to a double (and that makes sense).

I think we need the other version in case we pass in something that'll get autodiffed (fvars, or whatnot). I think what you have here is a suitable way to code the alternative.

The need for the alternative should definitely be doc'd in the function comments though.

Reply:
Cool beans feel free to chose this after docs are added